index.js

```diff
@@ -1,81 +1,120 @@
+console.log(name)
+//=> TypeError
+console.log("%c----------", "color: red") 
 
 
 
 // ***** Question 3 *****
 // *** Uncomment the lines below to test
-// console.log("%cQuestion 3", "color: red") 
+ console.log("%cQuestion 3", "color: red") 
+
+ function drinkWater(current_thirst_level){
+  console.log ("Man I sure am thirsty")
+  current_thirst_level -= 1
+  console.log ("Ahh that hits the spot")
+  return current_thirst_level
+ }
 
-// console.log(drinkWater(12))
+ console.log(drinkWater(12))
 // // => "Man I sure am thirsty"
 // // => "Ahh that hits the spot"
 // // => 11
-// console.log("%c----------", "color: red") 
+ console.log("%c----------", "color: red") 
 
 
 
 // ***** Question 4 *****
 // *** Uncomment the lines below to test
-// console.log("%cQuestion 4", "color: red") 
+ console.log("%cQuestion 4", "color: red") 
+
+function sameSameButDifferent(num, maybeNum){
+  if (num === maybeNum){
+   return'same same'
+  }
+ else if  (num == maybeNum) {
+  return 'same same but different'
+  }
+  else{
+    return 'not same'
+  }
+}
 
-// console.log(sameSameButDifferent(5, 5)) 
-// // => "same same"
 
-// console.log(sameSameButDifferent(123, "123")) 
+
+ console.log(sameSameButDifferent(123, "123")) 
 // // => "same same (but different)"
 
-// console.log(sameSameButDifferent(5, 7)) 
+ console.log(sameSameButDifferent(5, 7)) 
 // // => "different"
-
-// console.log(sameSameButDifferent(123, "122")) 
+console.log(sameSameButDifferent(123, 123)) 
 // // => "different"
-// console.log("%c----------", "color: red") 
+console.log("%c----------", "color: red") 
 
 
 
 // ***** Question 5 *****
 // *** Uncomment the lines below to test
-// console.log("%cQuestion 5", "color: red")
-// const student1 = { name: "Duane", grade: 88 }
-// updateGrade(student1, 92)
-// console.log(student1)
+ console.log("%cQuestion 5", "color: red")
+
+ 
+ const student1 = { name: "Duane", grade: 88 }
+
+ function updateGrade(student, grade){
+  student.grade = grade 
+ }
+ updateGrade(student1, 92)
+
+ console.log(student1)
 // // => { name: "Duane", grade: 92 }
 // console.log("%c----------", "color: red")
 
 
 
 // ***** Question 6 *****
 // *** Uncomment the lines below to test
-// console.log("%cQuestion 6", "color: red")
-
-// const users = [ 
-//   { 
-//     name: "Duane", phones: { cell: "555-123-4567", office: "555-456-7890" }
-//   },
-//   { 
-//     name: "Liza", phones: { cell: "555-234-5678", office: "555-567-1234" }
-//   }
-// ]
-// printNameAndPhones(users)
+ console.log("%cQuestion 6", "color: red")
+
+ const users = [ 
+  { 
+    name: "Duane", phones: { cell: "555-123-4567", office: "555-456-7890" }
+  },
+  { 
+    name: "Liza", phones: { cell: "555-234-5678", office: "555-567-1234" }
+  }
+]
+
+ function printNameAndPhones(users){
+  users.forEach(function(value){
+   console.log (value.name)
+   console.log (`cell ${value.phones.cell}`)
+   console.log (`office ${value.phones.office}`)
+  })
+ 
+  
+}
+
+
+printNameAndPhones(users)
+
+
```

Comment:
Looks solid. Nice job transforming things we were comfortable with in Ruby to their JS equivalent.

- .each  (vs) .forEeach
- bracket notation (vs) object dot notation
- "#{}" (vs) `${}`